File: js/number1.js
const m = 3;
const n = 3;
const s = "ULDDRRUU";

function startPosFinder(m, n, s) {
    let x = 0, y = 0;
    let rX = 0, lX = 0, uY = 0, dY = 0;

    for (const i of s) {
        if (i === "D") {
            y += 1;
            if (y > dY) dY = y;
        } else if (i === "U") {
            y -= 1;
            if (y < uY) uY = y;
        } else if (i === "L") {
            x -= 1;
            if (x < lX) lX = x;
        } else if (i === "R") {
            x += 1;
            if (x > rX) rX = x;
        }
    }

    if (rX - lX >= m || dY - uY >= n) {
        return "No";
    }
    return [1 - lX, 1 - uY];
}

console.log(startPosFinder(m, n, s));

File: rs/number1.rs
fn start_pos_finder(m: i32, n: i32, s: &str) -> String {
    let (mut x, mut y) = (0, 0);
    let (mut r_x, mut l_x) = (0, 0);
    let (mut u_y, mut d_y) = (0, 0);

    for c in s.chars() {
        match c {
            'D' => {
                y += 1;
                if y > d_y {
                    d_y = y;
                }
            }
            'U' => {
                y -= 1;
                if y < u_y {
                    u_y = y;
                }
            }
            'L' => {
                x -= 1;
                if x < l_x {
                    l_x = x;
                }
            }
            'R' => {
                x += 1;
                if x > r_x {
                    r_x = x;
                }
            }
            _ => {}
        }
    }

    if r_x - l_x >= m || d_y - u_y >= n {
        return "No".to_string();
    }
    format!("{} {}", 1 - l_x, 1 - u_y)
}

fn main() {
    let m = 3;
    let n = 3;
    let s = "ULDDRRUU";
    println!("{}", start_pos_finder(m, n, s));
}